File: snippets/cpp/02_sorting/selection_sort.cpp
#include <algorithm>
#include <vector>

std::vector<int> selection_sort(std::vector<int> arr) {
    int n = static_cast<int>(arr.size());

    for (int i = 0; i < n; ++i) {
        int min_idx = i;
        for (int j = i + 1; j < n; ++j) {
            if (arr[j] < arr[min_idx]) min_idx = j;
        }
        std::swap(arr[i], arr[min_idx]);
    }

    return arr;
}

File: snippets/rust/02_sorting/selection_sort.rs
pub fn selection_sort(input: &[i32]) -> Vec<i32> {
    let mut arr = input.to_vec();
    let n = arr.len();

    for i in 0..n {
        let mut min_idx = i;
        for j in (i + 1)..n {
            if arr[j] < arr[min_idx] {
                min_idx = j;
            }
        }
        arr.swap(i, min_idx);
    }

    arr
}
